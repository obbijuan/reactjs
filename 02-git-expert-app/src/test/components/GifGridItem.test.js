import React from 'react'
const { shallow } = require("enzyme")
const { GifGridItem } = require("../../components/GifGridItem")

describe('Pruebas en <GifGridItem>', () => {
    
    const title = "Titulo Ejemplo";
    const url = "https://localhost/test.jpg"
    const wrapper = shallow(<GifGridItem title={title} url={url}/>)

    test('debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de tener un parrafo con el title', () => {
        const p = wrapper.find('p').text().trim()
        expect(p).toBe(title)
    })

    test('debe tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img')
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })
    
    test('debe tener la clase "card".', () => {
        const clase = 'card'
        const div = wrapper.find('div')
        const className = div.prop('className')
        expect(className.includes(clase)).toBe(true);
    })
    
    
    
})
