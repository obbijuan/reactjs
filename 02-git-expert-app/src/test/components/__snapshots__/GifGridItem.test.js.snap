// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem> debe de mostrar el componente correctamente 1`] = `
<div
  className="card"
>
  <img
    alt="Titulo Ejemplo"
    src="https://localhost/test.jpg"
  />
  <p>
    Titulo Ejemplo
  </p>
</div>
`;
